consensus/reactor.go

```diff
@@ -794,6 +793,38 @@ OUTER_LOOP:
 	}
 }
 
+func (conR *ConsensusReactor) peerStatsRoutine() {
+	for {
+		// Manage disconnects from self or peer.
+		if !conR.IsRunning() {
+			conR.Logger.Info("Stopping peerStatsRoutine")
+			return
+		}
+
+		select {
+		case msg := <-conR.conS.statsMsgQueue:
+			// Get peer
+			peer := conR.Switch.Peers().Get(msg.PeerID)
+			if peer != nil {
```

Comment:
Good point. It could happen that at the point peerInfo arrives the peer is removed; I guess we can't do much in that case.

Reply:
I don't think we should reset in on disconnect as this is more consensus level stats. It does not care if underlying connection is temporarily broken or not; this could be important metric for peer quality but it's different concern.